citadel-internal-service-types/src/lib.rs

```diff
@@ -653,6 +661,102 @@ pub enum InternalServiceResponse {
     ListRegisteredPeersFailure(ListRegisteredPeersFailure),
 }
 
+/// Shortcut for getting `.request_id` attribute on all members
+macro_rules! match_request_id {
+    ($val:expr, $($variant:ident),+) => {
+        match $val {
+            $(
+                InternalServiceResponse::$variant(x) => x.request_id,
+            )+
+        }
+    }
+}
+
+impl InternalServiceResponse {
```

Comment:
I think you could use a proc macro to simplify this. Take a look at the answer on [this](https://stackoverflow.com/questions/68025264/how-to-get-all-the-variants-of-an-enum-in-a-vect-with-a-proc-macro).